### PR TITLE
fix(training): refuse an on-policy trust-region half-width that cannot be honored

### PR DESCRIPTION
**Pull request by cagataycali**
## Problem

`RLTrainSpec.clip_param` is the half-width of the trust region PPO is named for. `PpoTrainer.update` reads it twice per mini-batch, in the two expressions that clip:

```python
surrogate_clipped = -adv * torch.clamp(ratio, 1.0 - spec.clip_param, 1.0 + spec.clip_param)
value_clipped = old_values + (value - old_values).clamp(-spec.clip_param, spec.clip_param)
```

Nothing judged the value, and `torch.clamp` cannot: it is defined for every value below, so each one produced a **finite, successful, deployable** run whose objective was not the one the caller configured.

Measured on the on-policy backend over a seeded 60-step SO-100 reach run, against a never-trained control whose checkpoint parameter sum is `139.8929914773252676` and an honored default of `140.1741519418580992`:

| `clip_param` | `validate()` | outcome | checkpoint parameter sum |
|---|---|---|---|
| `0.2` (shipped default) | accepted | clips as configured | `140.1741519418580992` |
| `inf` | accepted | no clip, finite losses | `140.1735330768706262` |
| `nan` | accepted | **bit-identical to the unclipped run**, every loss `nan` | `140.1735330768706262` |
| `True` | accepted | silent half-width of one | `140.1735330768706262` |
| `-0.2` | accepted | clamp bounds inverted to a constant | `140.1913412402318500` |
| `0` | accepted | value clip pinned to `old_values` | `140.2282075245283863` |
| `-inf` | accepted | trains on `inf` losses | `140.0613218537641842` |
| `"0.2"` / `None` / `[0.2]` | accepted | raise `TypeError` at `rl/ppo.py:443` mid-update | -- |

Three rows are why this is a gate rather than a lint.

**A `nan` half-width silently removes the trust region.** Both clipped terms become `nan`, so `torch.max(surrogate, surrogate_clipped)` returns `nan` -- but its *gradient* flows to the **unclipped** branch, because every comparison against `nan` is false. The run therefore descends the unclipped objective, and its checkpoint comes out bit-identical to the `inf` run, while `surrogate_loss`, `value_loss` and `latest_loss` are all reported as `nan`. PPO's defining mechanism is off, and the only signal that anything happened is a metric a caller cannot act on.

**A negative half-width is not a window.** `1 - c` exceeds `1 + c`, so the clamp bounds are inverted and it returns a constant regardless of the ratio -- `clamp([0.7, 1.0, 1.4], 1.2, 0.8)` is `[0.8, 0.8, 0.8]` -- and the reported surrogate loss changes sign, from `-0.008662` to `+0.081992`. `0` is the same failure at the boundary: the value clip becomes `clamp(-0, 0)`, so `value_clipped` is exactly `old_values` and the critic's clipped branch is a constant.

**A `bool` is a silently different trust region.** `bool` is an `int` subclass, so `clip_param=True` is a half-width of one -- five times the shipped `0.2` -- written by a value that reads as a flag.

## Change

`PpoTrainer.validate` now reports a half-width that is not a positive real, so an unusable value is refused before the environment, the networks or the optimizer are built. `train()` fails closed on `validate()`, so the refusal replaces the deep stack trace a read-only preflight exists to prevent.

**Positive infinity stays inside the domain.** It is the field's only spelling of *do not clip*, and the consumer honors it: `clamp(ratio, -inf, inf)` returns the ratio unchanged, which is a coherent unclipped run with finite losses. A guard that refused a value its own consumer applies coherently would be narrower than the code it protects.

**One rule, one home.** That is the same domain, for the same reason, as the sibling clip bound `max_grad_norm`. Rather than add a second copy free to drift from the first, the existing private helper is **renamed in place** (`_gradient_clip_error` -> `_clip_bound_error`) and read by both gates, with its docstring covering both consumers. The rename is mechanical: one production call site, two docstring cross-references, six references in the sibling test module -- symbol names only, no assertion changed.

`clip_range_problems` is scoped like `gae_lambda_problems`: `spec.clip_param` appears in `rl/ppo.py` and nowhere else, so the off-policy backend stays silent about a field it never reads. A structural guard derives the reader set from the tree, so a second backend that starts clipping with the field fails until it routes through the gate.

The `discount_factor_problems` docstring listed `clip_param` as out of scope because it "needs the endpoint decision `gradient_clip_problems` records for the sibling bound". That decision now exists in the tree, so the caveat is corrected in the same change.

## Measured

On-policy clip-range domain

Every cell is read from two JSON dumps produced by one script run in a `git worktree` at `main` and on this branch; the generator asserts each rendered number, that the two dumps came from different trees, and that `nan`'s checkpoint sum is byte-equal to `inf`'s before it will save.

Values honored that no reading makes usable: **main 8 of 8 -> this change 0 of 8**, with all 3 usable values still accepted.

**The configured run is untouched, byte for byte.** Trained on both trees with the shipped default, the checkpoint parameter sum is `140.1741519418580992` on both, 40 deterministic steps of the resulting policy leave the SO-100 `Elbow` at `0.199985` rad on both, and the headless MuJoCo render of that pose differs by `max|delta| = 1/255` -- renderer noise. `np.float64(0.2)` is accepted and bit-identical to `0.2`, so a value read from a config array still trains.

## Scope

`init_noise_std` is the one field left out of the six optimization gates, for a measured reason rather than by omission: every non-finite value is already refused by `torch`, which rejects a `Normal` of non-positive or non-finite scale. The `discount_factor_problems` caveat is updated to say so.

## Verification

Run on the branch head `5faf58a`, based on `1610083`:

- `MUJOCO_GL=egl python -m pytest tests -q --no-cov` -> **24109 passed, 257 skipped, 0 failed** in 533 s.
- `ruff check strands_robots tests tests_integ` -> clean; `ruff format --check` -> 1121 files already formatted; `mypy strands_robots tests tests_integ` -> **0 errors outside `examples/isaac_gs`** (the 14 there are byte-identical to a pristine `main` worktree of the same working copy, so they are environmental).
- `python3 scripts/assemble_changelog.py --check` -> OK (236 pending); `scripts/check_merge_base_overlap.py` -> no overlap.
- **Pre-fix proof.** Reverting only the call sites (`training/base.py`, `training/rl/ppo.py`) to `main` while keeping `_validate.py`, so the test module still imports: **34 failed, 104 passed**. The structural guard's failure names `ppo.py` directly. The 104 that pass are the gate's own direct-call tests, the shared-domain parity and the `torch` premise measurements -- none of which depend on the wiring, which is what makes the 34 the wiring's own proof.
- `tests/training/` -> 2297 passed, 4 skipped. The 138 new tests are additive: the only change to an existing test file is the six renamed references to the shared helper.

### CODE REVIEW

**Review by yinsong1986 — APPROVED** [review]
## Summary

Adds a `validate()`-time domain gate for `RLTrainSpec.clip_param` on the on-policy backend, following the same pattern as the six sibling optimization gates (`max_grad_norm`, `gamma`, `lam`, epochs, loss weights): a half-width that is not a positive real is refused before the environment, networks, or optimizer are built, with `+inf` deliberately inside the domain as the field's only spelling of "do not clip" — a value the consumer (`torch.clamp`) honors coherently. Rather than duplicating the rule, the existing private `_gradient_clip_error` helper is renamed in place to `_clip_bound_error` and shared by both clip-bound gates; I verified the rename is complete across the tree (no stray references, helper never exported in `__all__`, so no public-API impact). The gate is correctly scoped: `spec.clip_param` is read only in `rl/ppo.py`, the off-policy backends stay silent about it, and a structural test derives the reader set from the tree so a future backend that starts clipping fails until it routes through the gate. Independently verified at the head SHA in a fresh worktree: the two affected test modules pass (338 passed), `tests/training/` passes with only two failures that reproduce identically on the base commit (pre-existing, environmental — the PR touches nothing in lerobot), and `ruff check` / `ruff format --check` are clean on all changed files.

## What's good

- Changelog fragment (`changelog.d/2047-clip-range-domain.md`) per AGENTS.md PR workflow step 3; no direct `CHANGELOG.md` edit.
- Fail-closed with a named, actionable error message replacing a silent misconfigured run — the exact no-silent-defaults property AGENTS.md requires; the `nan` case (trust region silently off, checkpoint bit-identical to unclipped) is a genuinely dangerous silent failure and the measurement table demonstrates it.
- Shared-domain parity tests pin the two clip bounds to one rule so they cannot drift, and the structural one-owner scan guards the scoping decision.
- No host paths, no emojis, no new non-ASCII in the diff; the only change to an existing test file is the six mechanical symbol renames, with no assertion changed.
